association_backend/association/domain/entities/subscription_entities.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+import pydantic
+from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Table
+from sqlalchemy.orm import registry, relationship
+
+
+class UserData(pydantic.BaseModel):
+    email: str
+    user_id: int
+
+
+class SubscriptionState(str, Enum):
+    PENDING = "pending"
+    ACTIVE = "active"
+    EXPIRED = "expired"
+    NOT_CREATED = "not-created"
+
+    def __str__(self) -> str:
+        return str.__str__(self)
+
+
+@dataclass
+class Subscription:
+    user_id: int
+    creation_date: datetime
+    state: SubscriptionState
+    # is_for_life: bool
+    user_email: str = ""
+    stripe_session_id: Optional[str] = ""
+    due_date: Optional[datetime] = None
+    stripe_id: Optional[str] = ""
+    stripe_customer_id: Optional[str] = ""
+
+
+@dataclass
+class SubscriptionPayment:
+    subscription: Subscription
+    payment_date: datetime
+    invoice_id: str
+    invoice_pdf: str
+
+
+# =============
+mapper_registry = registry()
+
+subscription_table = Table(
+    "subscription",
+    mapper_registry.metadata,
+    Column("user_id", Integer(), nullable=False, primary_key=True),
+    Column("user_email", String(128), nullable=True),
+    Column("creation_date", DateTime(timezone=True), nullable=False),
+    Column("next_payment_due_date", DateTime(timezone=True), nullable=True),
+    Column("stripe_id", String(128), nullable=True),
```

Comment:
What does this id represent? others have customer/session, why this is just `stripe_id`?

Reply:
`stripe_id` is` stripe_subscription_id`
I named it that way because we are in the Subscription Model